### PR TITLE
OPENDNSSEC-786 align cmdhandler code

### PR DESCRIPTION
**Pull request by halderen**

### CODE REVIEW

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
There are things left to do/decide:
- fixed/variable number of command threads
- detached/join threads
- interaction cmd line program <> daemon
- command line arguments parse/tokenizing?
- Keep handle function?
- usage()/help() as separate functions or just text.

**Comment by halderen** [comment]
#build

File: enforcer/src/daemon/engine.h [hunk 1]
```diff
@@ -76,7 +72,7 @@ struct engine_struct {
  * \param dbcfg_list, database configuration list
  * \return connection on success, NULL on failure.
  */
-db_connection_t* get_database_connection(db_configuration_list_t* dbcfg_list);
+db_connection_t* get_database_connection(engine_type* dbcfg_list);
```
Comment (halderen) [inline enforcer/src/daemon/engine.h]: Ack.

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
#build

**Comment by halderen** [comment]
@yschaeff @hodarh review and merge ready

**Comment by yschaeff** [comment]
Seems correct

**Pull request merged**